packages/react-dom/src/__tests__/ReactDOMEventListener-test.js

```diff
@@ -191,8 +211,80 @@ describe('ReactDOMEventListener', () => {
         // change anyway. We can maybe revisit this later as part of
         // the work to refine this in the scheduler (maybe by leveraging
         // isInputPending?).
-        expect(mock.mock.calls[1][0]).toBe('1');
-        // By the time we leave the handler, the second update is flushed.
+        //
+        // Since this is a discrete event, the previous update is already done.
+        if (gate(flags => flags.enableLegacyFBSupport)) {
```

Comment:
I don't think this was the right fix, I think the issue is we're not dispatching up to the document, because the other tests are failing for the same reason. I'll look deeper and figure out the fix

Reply:
I think the issue here isn't the document. I'm not sure how this works exactly if it's two listeners at the document or just one.

I think it's actually two listeners at the document. So we should flush between them.

The issue is that `dispatchAndWaitForDiscrete` waits between nodes but it doesn't wait between events on the same node.

I wonder if there's a way to get to the internal list of events in JSDOM. Otherwise, you might have to patch `EventTarget.prototype.addEventListener` so you can gather your own list and iterate through it and wait between each one.

Reply:
Looks like you can get access to internals through the symbol:

```
const impl = Object.getOwnPropertySymbols(x)[0];
x[impl]._eventListeners
```

They should've called it DO_NOT_USE.

You could potentially fork https://github.com/jsdom/jsdom/blob/2f8a7302a43fff92f244d5f3426367a8eb2b8896/lib/jsdom/living/events/EventTarget-impl.js#L114 and make it an async function.